internal/services/validate_user.go

```diff
@@ -0,0 +1,34 @@
+package services
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/gofiber/fiber/v2/log"
+	"github.com/golang-jwt/jwt"
+	"github.com/spf13/viper"
+)
+
+func (s *Server) BearerAuth(bearerHeader string) (interface{}, error) {
```

Comment:
Думаю нет смысла использовать `interface{}`
```suggestion
func (s *Server) BearerAuth(bearerHeader string) (string, error) {
```